### PR TITLE
Traffic Router upgrade Tomcat to 8.5

### PR DESCRIPTION
**Pull request by ajschmidt**
This PR should not be accepted in the 'master' branch at this time. This is a major upgrade to Traffic Router to be released with Traffic Control 3.0, or at least that is the current plan. The PR includes the following changes:

1. Replaced custom Java SNI implementation with a native implementation using tomcat-native, apr (Apache Portable Runtime) and OpenSSL This should significantly improve the performance of routing ‘https’ delivery services.
2. Upgraded to Tomcat 8.5.30
3. Separated the Traffic Router installation from the Tomcat deployment and created a new ‘tomcat’ package for installing Tomcat. Traffic Router and Tomcat can now be upgraded independently
4. Converted Traffic Router to a ‘systemd’ service
5. Modified the development test and dev deployment processes to be more consistent with production

For those using the RPM installation the installation process will not change much, but this version does require CentOS 7 or higher. The additional Tomcat RPM is created by the build process and will need to be added to an accessible RPM package library. 

For those deploying Traffic Router on a development desktop the changes will me much more significant. Because the OpenSSL, Tomcat Native and APR native libraries have to be installed and the Traffic Router build time and runtime need to know where to find those libraries using the '-Djava.library.path='. 

This upgrade allows Traffic Router to keep up with newer versions of Java and Tomcat. It also increases Traffic Router's capacity to handle more HTTPS delivery services.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by limited** [comment]
test this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1679/
Test PASSed.

**Comment by dangogh** [comment]
also see in the rat report -- this file doesn't have the Apache license header:
```
traffic_router/tomcat-rpm/tomcat.service
```

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.3.0
```
Comment (dangogh) [inline VERSION]: The VERSION change really must be a PR in itself (perhaps with documentation changes). 

Can you remove this change?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Pull request merged**